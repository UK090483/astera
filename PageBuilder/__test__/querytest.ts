import { parse, evaluate } from "groq-js";

export const testQuery = async (query: string, dataSet?: any[]) => {
  let _dataset = [
    { _type: "user", name: "Michael" },
    { _type: "company", name: "Bluth Company" },
  ];

  try {
    const tree = parse(query);
    let value = await evaluate(tree, { dataset: dataSet || _dataset });
    let result = await value.get();
    return result;
  } catch (error) {
    return false;
  }
};

export const testProjection = async (projection: string) => {
  try {
    return await testQuery(`*[_type == 'page'][]{${projection}}`);
  } catch (error) {
    return false;
  }
};
