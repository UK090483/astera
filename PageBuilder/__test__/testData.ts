import { SanityObjectDefinition } from "PageBuilder/types";

export const testData = {
  contentType: {
    one: { name: "contentType1_name", title: "contentType1_title" },
    two: { name: "contentType2_name", title: "contentType2_title" },
  },
  get object() {
    return {
      one: {
        name: "component1_name",
        title: "component1_title",
        fields: [this.field.text, this.field.string],
      } as Omit<SanityObjectDefinition, "type">,
      two: {
        name: "component2_name",
        title: "component2_title",
        fields: [this.field.text, this.field.string],
      } as Omit<SanityObjectDefinition, "type">,
    };
  },

  get field() {
    return {
      text: { name: "text", title: "text", type: "text" },
      string: { name: "string", title: "string", type: "string" },
      number: { name: "number", title: "number", type: "number" },
      boolean: { name: "boolean", title: "boolean", type: "boolean" },
      slug: { name: "slug", title: "slug", type: "slug" },
      link: { name: "link", title: "link", type: "link" },
      get array() {
        return { name: "array", title: "array", type: "array", of: [] };
      },
    };
  },
};
