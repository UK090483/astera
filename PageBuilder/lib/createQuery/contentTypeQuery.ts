import { defaultEmptyArray } from "../../helper";
import { Config, contentType } from "../../types";
import { resolveContentType } from "../sanity/createContentTypes";
import {
  fieldsToQuery,
  schemaItemToQuery,
  contentTypeQuery,
} from "./fieldsToQuery";

export const contentTypesQuery = (config: Config) => {
  return defaultEmptyArray(config.contentTypes).map((i) => {
    return contentTypeQuery(config, i);
  });
};

export const getContentTypeQuery = (config: Config, contentType: string) => {
  const _contentType = defaultEmptyArray(config.contentTypes).find(
    (i) => i.name === contentType
  );
  if (_contentType) {
    const q = contentTypeQuery(config, _contentType);

    let query = "_id,_type, " + q;

    if (config.hooks?.onContentTypeQuery) {
      query = config.hooks.onContentTypeQuery({
        config,
        result: query,
      });
    }

    return query;
  }
  console.warn("unable to find contentType for queryGeneration");
  return "";
};
