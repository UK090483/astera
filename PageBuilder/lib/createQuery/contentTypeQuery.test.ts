import { testData } from "../../__test__/testData";
import { getContentTypeQuery } from "./contentTypeQuery";

describe("getContentTypeQuery", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it("should warn if contentType not found", () => {
    console.warn = jest.fn();
    const q = getContentTypeQuery(
      { contentTypes: [testData.contentType.one] },
      "bla"
    );
    expect(console.warn).toBeCalledTimes(1);
  });

  it("should create initial Query", () => {
    const q = getContentTypeQuery(
      { contentTypes: [testData.contentType.one] },
      testData.contentType.one.name
    );
    expect(q).toMatchSnapshot();
  });

  it("should create initial Query with locale", () => {
    const q = getContentTypeQuery(
      { contentTypes: [testData.contentType.one] },
      testData.contentType.one.name,
      "en"
    );
    expect(q).toMatchSnapshot();
  });

  it("should add components to query", () => {
    const q = getContentTypeQuery(
      {
        contentTypes: [testData.contentType.one],
        components: [testData.object.one],
      },
      testData.contentType.one.name,
      "en"
    );
    expect(q).toMatchSnapshot();
  });

  it("should handle on create Query", () => {
    const q = getContentTypeQuery(
      {
        contentTypes: [testData.contentType.one],
        hooks: {
          onContentTypeQuery: ({ result }) => {
            return `${result} test,`;
          },
        },
      },
      testData.contentType.one.name,
      "en"
    );
    expect(q).toMatchSnapshot();
  });
});
