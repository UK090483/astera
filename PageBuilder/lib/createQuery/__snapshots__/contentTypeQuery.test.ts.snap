// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getContentTypeQuery should add components to query 1`] = `"_id,_type, 'title': coalesce(title_en,title) ,'description': coalesce(description_en,description) ,'featuredImage': featuredImage{undefined} ,'slug': coalesce(slug_en,slug).current, "`;

exports[`getContentTypeQuery should create initial Query 1`] = `"_id,_type, title ,description ,'featuredImage': featuredImage{undefined} ,'slug': slug.current, "`;

exports[`getContentTypeQuery should create initial Query with locale 1`] = `"_id,_type, 'title': coalesce(title_en,title) ,'description': coalesce(description_en,description) ,'featuredImage': featuredImage{undefined} ,'slug': coalesce(slug_en,slug).current, "`;

exports[`getContentTypeQuery should handle on create Query 1`] = `"_id,_type, 'title': coalesce(title_en,title) ,'description': coalesce(description_en,description) ,'featuredImage': featuredImage{undefined} ,'slug': coalesce(slug_en,slug).current,  test,"`;
